product_packaging_unit_price_calculator/wizards/product_package_price.py

```diff
@@ -0,0 +1,136 @@
+    )
+    product_supplierinfo_id = fields.Many2one(
+        "product.supplierinfo",
+        default=lambda self: self._default_product_supplierinfo_id(),
+    )
+    product_variant_ids = fields.One2many(
+        "product.product",
+        "product_tmpl_id",
+        related="product_tmpl_id.product_variant_ids",
+    )
+    selected_packaging_id = fields.Many2one(
+        "product.packaging",
+        domain="[('product_id', 'in', product_variant_ids)]",
+    )
+    packaging_price = fields.Float("Package Price", default=0.0, digits="Product Price")
+    unit_price = fields.Float(
+        compute="_compute_unit_price",
+        readonly=True,
+        digits="Product Price",
+    )
+    current_unit_price = fields.Float(
+        compute="_compute_current_unit_price", digits="Product Price"
+    )
+    packaging_ids = fields.One2many(
+        "product.packaging",
+        string="Product Packages",
+        compute="_compute_packaging_ids",
+    )
+    warning_message = fields.Char(readonly=True, default=" ")
+
+    @api.depends("packaging_price", "selected_packaging_id")
+    def _compute_unit_price(self):
+        if not self.selected_packaging_id:
+            self.unit_price = self.current_unit_price
+        elif not self.selected_packaging_id.qty:
+            self.unit_price = self.current_unit_price
+            self.warning_message = _(
+                "Unit price cannot be computed because the selected"
+                "packaging has no quantity set."
+            )
+        else:
+            self.unit_price = self.packaging_price / self.selected_packaging_id.qty
+            self.warning_message = " "
+        self._compute_package_prices()
+
+    @api.depends(
+        "product_pricelist_item_id", "product_supplierinfo_id", "product_tmpl_id"
+    )
+    def _compute_current_unit_price(self):
+        """Compute the original unit price, the one  that the calculator will change."""
+        if self.product_pricelist_item_id:
+            self.current_unit_price = self.product_pricelist_item_id.fixed_price
+        elif self.product_supplierinfo_id:
+            self.current_unit_price = self.product_supplierinfo_id.price
+        elif self.product_id:
+            self.current_unit_price = self.product_id.lst_price
+        else:
+            self.current_unit_price = self.product_tmpl_id.list_price
+
+    @api.depends("unit_price")
+    def _compute_package_prices(self):
+        for pack in self.packaging_ids:
+            pack.packaging_wizard_price = self.unit_price * pack.qty
+
+    @api.depends("product_tmpl_id")
+    def _compute_packaging_ids(self):
+        self.packaging_ids = self.product_tmpl_id.mapped(
+            "product_variant_ids.packaging_ids"
+        )
+
+    def action_set_price(self):
+        if not self.packaging_price:
+            return
+        if not self.selected_packaging_id.qty:
+            raise UserError(
+                _(
```

Comment:
18] When possible, it is better to rely on `Env._`: replace '_('text')' by 'self.env._('text')' and remove '_' from the import